Add configure and start MCP server guide (#5947)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/start-mcp-server/main/index.md

---
title: Configure, start, and test the Okta MCP server
meta:
  - name: description
    content: Set up your local environment variables and start the Okta Model Context Protocol (MCP) server to connect your AI agent to your Okta org.
---

Set up your local environment variables and start the Okta Model Context Protocol (MCP) server to connect your AI agent to your Okta org.

---

#### Learning outcomes

* Map your Okta app credentials to MCP environment variables.
* Start the Okta MCP server.

> * Verify the connection between the Okta MCP server and your Okta org.

Reviewer: ```suggestion
* Verify the connection between the Okta MCP server and your org.
```

> * Use natural language commands to perform admin tasks (such as, user, group, application, and policy management).

Reviewer: This part is a little unclear. User, group, etc. aren't tasks. Maybe this should say "such as **managing** users, groups .... etc."

#### What you need

* The [uv](https://docs.astral.sh/uv/getting-started/installation/) package manager.
* [Python 3.8](https://python.org/downloads) or higher.
* A local copy of the Okta MCP server project (see, [Install, and initialize the Okta MCP server](https://github.com/okta/okta-mcp-server) guide).
* The credentials from the [app authentication for Okta MCP server](/docs/guides/configure-mcp-authentication/main/) guide:

>   * Okta domain (`OKTA_ORG_URL`): Your Okta org URL (see [Find your Okta domain](/docs/guides/find-your-domain/main/)).

Reviewer: ```suggestion
  * Okta domain (`OKTA_ORG_URL`): Your Okta org URL. See [Find your Okta domain](/docs/guides/find-your-domain/main/).
```

         "command": "uv",
         "args": [
           "run",
           "--directory",
           "/path/to/okta-mcp-server",
           "okta-mcp-server"
         ],
         "env": {
           "OKTA_ORG_URL": "https://your-org.okta.com",
           "OKTA_CLIENT_ID": "your-client-id",
           "OKTA_SCOPES": "okta.users.read okta.groups.read",
           "OKTA_PRIVATE_KEY": "your-private-key-if-using-jwt",
           "OKTA_KEY_ID": "your-key-id-if-using-jwt"
         }
       }
     }
   }
   ```

## Other MCP Clients

Apply the configuration parameters shown in the previous sections to other MCP-compatible clients. Refer to your client's documentation for the specific configuration file location.

---

## Start the Okta MCP server

The steps to start the server vary by client. To start the Okta MCP server in VS Code with GitHub Copilot, follow these steps:

### VS Code

1. Open **GitHub Copilot Chat** in VS Code.
1. At the bottom right of the Copilot chat panel, click the **Tools** icon.
1. In the tools dialog at the top, search for **MCP Server: okta-mcp-server**.
1. Click the **settings** icon to open the MCP server configuration (`mcp.json`) file.

> 1. In the **servers** section, locate **okta-mcp-server** and click **Start** to activate the server.

Reviewer: ```suggestion
1. In the **Servers** section, locate **okta-mcp-server** and click **Start** to activate it.
```

> 1. Follow the prompts to authorize your device by entering the user code in your browser and sign in with your Okta credentials to complete the activation.

Reviewer: ```suggestion
1. Follow the prompts to authorize your device by entering the user code in your browser. Then sign in with your Okta credentials to complete the activation.
```

1. The server prompts you to enter the following configuration variables during the first run:

   | Variable | Description | Required |
   | :--- | :--- | :--- |
   | `OKTA_ORG_URL` | Your Okta tenant org URL (example: `https://integrator-1234567.okta.com`). | Yes |
   | `OKTA_CLIENT_ID` | The client ID copied from your Okta app. | Yes |
   | `OKTA_SCOPES` | Space-separated list of API scopes you granted to the app (example: `okta.users.read okta.groups.read`). **Note:** Only include scopes that you granted for your app. | Yes |
   | `OKTA_PRIVATE_KEY` | Your private key in PEM format (starts with `-----BEGIN PRIVATE KEY-----`). | Private key JWT only |
   | `OKTA_KEY_ID` | The key ID (KID) for your private key. | private key JWT only |

>    > **Note**: If using the device authorization grant, press **Enter** to skip the `OKTA_PRIVATE_KEY` and `OKTA_KEY_ID` prompts.

Reviewer: ```suggestion
   > **Note**: If you use the device authorization grant, press **Enter** to skip the `OKTA_PRIVATE_KEY` and `OKTA_KEY_ID` prompts.
```
